datastore/filesystem_datastore.nim

```diff
@@ -0,0 +1,159 @@
+import std/os
+
+import pkg/questionable
+import pkg/questionable/results
+import pkg/upraises
+
+import ./datastore
+
+export datastore
+
+push: {.upraises: [].}
+
+type
+  FileSystemDatastore* = ref object of Datastore
+    root: string
+
+const
+  objExt* = ".obj"
```

Comment:
.obj files are already used in Windows for compiled code (similar to .o in Unix). May be use another extension, such as .dsobject?

Reply:
Ah, great tip, didn't know that. `dsobject` works for me.

Reply:
Change made.